File: backend/Models/Adresse.cs
// Modèle de données pour une adresse (Classe utilisée pour représenter une adresse dans le système de gestion des activités communautaires)
namespace backend.Models;

public class Adresse
{
    public required string Numero { get; set; } // Numéro de rue de l'adresse
    public required string Rue { get; set; } // Rue de l'adresse
    public required string Ville { get; set; } // Ville de l'adresse
    public required string CodePostal { get; set; } // Code postal de l'adresse
    public required string Province { get; set; } // Province de l'adresse
}

File: frontend/proconnectnb/lib/models/adresse.dart
class Adresse {
  final String? rue;
  final String? ville;
  final String? codePostal;

  Adresse({this.rue, this.ville, this.codePostal});

  factory Adresse.fromJson(Map<String, dynamic> json) {
    return Adresse(
      rue: json['rue'],
      ville: json['ville'],
      codePostal: json['codePostal'],
    );
  }

  Map<String, dynamic> toJson() => {
    "rue": rue,
    "ville": ville,
    "codePostal": codePostal,
  };
}